File: bin/remap_azimuth_l2.R
#!/usr/bin/env Rscript

# __date__ <- '2020-06-26'
# __version__ <- '0.0.1'

library(data.table)

# Function to parse command line arguments
parse_args <- function() {
  args <- commandArgs(trailingOnly = TRUE)
  arg_list <- list()

  for (i in seq(1, length(args), by = 2)) {
    arg_name <- sub("--", "", args[i])
    arg_value <- args[i + 1]
    arg_list[[arg_name]] <- arg_value
  }

  # if ("version" %in% names(arg_list)) {
  #   cat("Version", __version__, "\n")
  #   quit(status = 0)
  # }

  if (!all(c("az_file", "mapping", "out_file") %in% names(arg_list))) {
    stop("Missing required argument(s). Required: --az_file, --mapping, --out_file", call. = FALSE)
  }

  return(arg_list)
}

# Parse arguments
args <- parse_args()

# Read data with error handling for compressed files

tryCatch({
  All_Data <- fread(args$az_file, sep = "\t", showProgress = FALSE)
  compressed <- TRUE
}, error = function(e) {
  All_Data <- fread(args$az_file, sep = "\t", showProgress = FALSE)
  compressed <- FALSE
})
# print(All_Data)
# print('loaded')
# Read mapping file
Mappings <- fread(args$mapping, sep = "\t", data.table = FALSE)
# print(Mappings)
# Data manipulation
D1 <- All_Data
D1$idx1 <- seq_len(nrow(D1))
setDT(D1)

# Perform the replacement
# We will create a new column for each of the columns in Mappings
print(D1)
for (col in colnames(Mappings)) {
  if (col == "L2") {
    next  # Skip this iteration if col is L2
  }
  D1[, paste0(col, '_predicted.celltype.l2') := Mappings[[col]][match(predicted.celltype.l2, Mappings$L2)]]
}
print(D1)
# Write to output file
fwrite(D1, file = args$out_file, sep = "\t", quote = FALSE)

cat("Done\n")

File: bin/remap_azimuth_l2.py
#!/usr/bin/env python

__date__ = '2020-06-26'
__version__ = '0.0.1'
import pandas as pd
import argparse
import os

parser = argparse.ArgumentParser(
    description="""
        Basic QC plots.
        """
)

parser.add_argument(
    '-v', '--version',
    action='version',
    version='%(prog)s {version}'.format(version=__version__)
)

parser.add_argument(
    '-az', '--az_file',
    action='store',
    dest='az_file',
    required=True,
    help='H5 AnnData file.'
    )

parser.add_argument(
    '-m', '--mapping',
    action='store',
    dest='mapping',
    required=True,
    help='H5 AnnData file.'
    )

parser.add_argument(
    '-of', '--out_file',
    action='store',
    dest='of',
    required=True,
    help='H5 AnnData file.'
    )

options = parser.parse_args()

All_Data = pd.read_csv(options.az_file,compression='gzip',sep='\t')
Mappings = pd.read_csv(options.mapping,sep='\t',index_col=0)

D1 =All_Data
D1['idx1']=D1.index
D1 = All_Data.set_index('predicted.celltype.l2')
for col in Mappings.columns:
    D1[f'{col}_predicted.celltype.l2']=''
    D1[f'{col}_predicted.celltype.l2']=Mappings[col]
D1 =D1.reset_index()
D1 = D1.set_index('idx1')
D1.to_csv(options.of,compression='gzip',sep='\t')    
print('Done')